src/extra.rs

```diff
@@ -26,6 +27,21 @@ pub type Default = Full<DefaultErr, DefaultState, DefaultCtx>;
 pub type Err<E> = Full<E, DefaultState, DefaultCtx>;
 
 /// Use specified state type, but default other types
+///
+/// Use `State<S>` or `Full<E, S, C>` as the `E` type parameter of a parser to use a custom state type.
```

Comment:
Do you mean `S` instead of `E`?

Reply:
I meant the Extra type parameter of Parser - I'll just say that instead to be more clear.